pkg/controller/dataimportcron-controller.go

```diff
@@ -900,7 +900,7 @@ func (r *DataImportCronReconciler) cleanup(ctx context.Context, cron types.Names
 func (r *DataImportCronReconciler) deleteJobs(ctx context.Context, cron types.NamespacedName) error {
 	deletePropagationBackground := metav1.DeletePropagationBackground
 	deleteOpts := &client.DeleteOptions{PropagationPolicy: &deletePropagationBackground}
-	selector, err := getSelector(map[string]string{common.DataImportCronLabel: getCronJobLabelValue(cron.Namespace, cron.Name)})
+	selector, err := getSelector(map[string]string{common.DataImportCronLabel: GetCronJobLabelValue(cron.Namespace, cron.Name)})
```

Comment:
I think a little further down https://github.com/kubevirt/containerized-data-importer/blob/98d0caeb56b7fc23963eeca2e3db8329f8506d23/pkg/controller/dataimportcron-controller.go#L907-L910 can be replaced with a direct delete? I may be wrong but seems to be equivalent

Reply:
How come it's needed for jobs?

Reply:
There might be several jobs (according to the cronjob spec: 1 Active, 1 Successful, 1 Failed) from the cronjob, and also the initial job.

Reply:
On second thought, we have TTLSecondsAfterFinished set (10sec) for all jobs, so they will be deleted anyway and not sure we must delete them explicitly earlier.

Reply:
@akalenyu regarding the cronjob deletion loop, since we may get there after the DIC is already deleted (like in the bug), we cannot delete the cronjob by name as it's based on the DIC UID
https://github.com/kubevirt/containerized-data-importer/blob/42ec627e3593c45027c2ebffb32f26a182c9cdee/pkg/controller/dataimportcron-controller.go#L1337-L1340
so unfortunately I think we should keep this loop.

Reply:
Not sure I understand the problem. We can always convert the label value on the cron job to the deleted cron's name,
which is all we need to use DeleteAllOf with a label selector. No?

Reply:
Sure, we can use DeleteAllOf (fixing it), but note the convertion mentioned is currently buggy as the label may truncate the name
https://github.com/kubevirt/containerized-data-importer/blob/42ec627e3593c45027c2ebffb32f26a182c9cdee/pkg/controller/dataimportcron-controller.go#L1351-L1358

Reply:
No. As said, the List and Delete loop is replaced with DeleteAllOf. The bug in this originial PR is in the other direction: converting the watched cronjob label (which is possibly truncated) to reconcile ns/name (of its DIC). Fixing it as well.